### DISCUSSION TITLE
More control over when customers are required to log in

### DISCUSSION THREAD

**Discussion by jeriox** [pretix/pretix#3980] [Ideas]
We are building a solution where we require customers to log in to their customer accounts in pretix to be able to do some checks. This results in a mediocre UX for customers, mainly because of the way CustomerStep is currently implemented. We had two ideas to improve this:

1. (preferred): Add a setting whether to show the CustomerStep. Currently, this is tied to whether the customer accounts feature is enabled. This way, we could hide the step for the majority of users who should not be required to log in (making the checkout faster) and handle the login requirement in a custom step if applicable. 
2. Mark products as "login required". This could lead to similar behaviour as with memberships, where the CustomerStep enforces a login

**Comment by raphaelm** [comment]
This has some overlap with #3980 which introduces an option to require users to create a customer account. I'd be open to doing that on event level, I'm not sure I'd be open to do it on product level.

Adding an option to not showing the customer login even though it is technically enabled feels like something way out of scope for pretix, sorry. This is an understandable, but very unique use case and we try to build a very general system and need to keep complexity manageable.

**Reply by jeriox** [reply]
You referenced this issue in your comment 😅 

In our opinion it complicates/prolongs the checkout flow quite a lot to have a whole step dedicated to the login, especially because we also sell single small items like drinks which the user is not willing to spend a lot of time in the online shop for. We like the decent "log in" in the upper right corner and maybe a subtle hint next to the form for the mail address like "Log in and use saved data from your account". But its also totally understandable if that is too unique to be included in core. (Would a signal be a possibility to determine whether to show the view/whether to require a login? That would keep the UI simple but allow us to input on that decision)

**Reply by BenBals** [reply]
Would adding this as a signal be an option? This signal could receive a cart and decide whether to show the login step to users (and/or force log in).

**Reply by raphaelm** [reply]
Sorry, I meant to reference #2991 :joy: 

I'll discuss this internally a little!

**Reply by raphaelm** [reply]
We've discussed this and believe a signal is a bad idea. It would break in bad ways e.g. when combined with products that require a membership. We re not entirely opposed to a setting, but also don't really think it's worth it.

**Reply by jeriox** [reply]
I can see how that could break, but couldn't the signal response be treated accordingly? (e.g. if a plugin asks for the login to be hidden, but a product requires a memberships, the login would still be shown) kind of indicating a preference for the login if that is possible under the given circumstances
